synapseclient/models/wiki.py

```diff
@@ -1 +1,1278 @@
+            else:
+                raise
+        else:
+            if not self.id:
+                raise ValueError("Must provide id to update existing wiki page.")
+            return "update_existing_wiki_page"
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Store the wiki page: {self.owner_id}"
+    )
+    async def store_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "WikiPage":
+        """Store the wiki page. If there is no wiki page, a new wiki page will be created.
+        If the wiki page already exists, it will be updated.
+        Subwiki pages are created by passing in a parent_id.
+        If a parent_id is provided, the wiki page will be created as a subwiki page.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The created/updated wiki page.
+
+        Raises:
+            ValueError: If owner_id is not provided or if required fields are missing.
+
+        Example: Store a wiki page
+            This example shows how to store a wiki page.
+            from synapseclient import Synapse
+            from synapseclient.models import (
+                Project,
+                WikiPage,
+            )
+            syn = Synapse()
+            syn.login()
+            project = await Project(name="My uniquely named project about Alzheimer's Disease").get_async()
+            wiki_page = await WikiPage(owner_id=project.id, title="My wiki page").store_async()
+            print(wiki_page)
+        """
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        wiki_action = await self._determine_wiki_action(synapse_client=client)
+        # get the markdown file handle and attachment file handles if the wiki action is valid
+        if wiki_action:
+            # Update self with the returned WikiPage objects that have file handle IDs set
+            self = await self._get_markdown_file_handle(synapse_client=client)
+            self = await self._get_attachment_file_handles(synapse_client=client)
+
+        # Handle root wiki page creation if parent_id is not given
+        if wiki_action == "create_root_wiki_page":
+            client.logger.info(
+                "No wiki page exists within the owner. Create a new wiki page."
+            )
+            # Create the wiki page
+            wiki_data = await post_wiki_page(
+                owner_id=self.owner_id,
+                request=self.to_synapse_request(),
+                synapse_client=client,
+            )
+            client.logger.info(
+                f"Created wiki page: {wiki_data.get('title')} with ID: {wiki_data.get('id')}."
+            )
+        if wiki_action == "update_existing_wiki_page":
```

Comment:
nit: There can be only a single `wiki_action`, so I could place these into an `elif` statement